plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/ConstraintChecks.kt

```diff
@@ -445,9 +445,14 @@ private fun SolverState.checkCallExpression(
               val descriptor = resolvedCall.resultingDescriptor
               if (descriptor.isField()) {
                 val fieldConstraint = solver.ints {
-                  equal(
-                    solver.makeObjectVariable(associatedVarName),
-                    solver.field(descriptor.fqNameSafe.asString(), solver.makeObjectVariable(argVars[0].second))
+                  val typeName = descriptor.fqNameSafe.asString()
+                  val argName = argVars[0].second
+                  NamedConstraint(
+                    "$associatedVarName == $typeName($argName)",
```

Comment:
Should we maybe also pass the `KtExpression` comes from so that we can use it instead of the `$associatedVarName`? This would make things clearer, at least for the places where names are completely made up.

Reply:
Agreed, see https://github.com/arrow-kt/arrow-meta/pull/850#discussion_r698789532